### DISCUSSION TITLE
Postprocessor to get coordinates from node id

### DISCUSSION THREAD

**Discussion by lyyc199586** [idaholab/moose#24655] [Q&A Modules: General] [answered]
Hi all,

Is there happen to be a postprocessor that can get coordinates form the node id? For example, I can use `NodalMaxValueId` to get the node id I want, is there a way to output the coords of this node?

**Comment by GiudGiud** [comment]
Hello

you can use paraview with min/max filters on the node id field to locate that particular node.
Do you only have one node you care about?

you could make a new Positions object and this will output the location of the node that this object selects to JSON
https://mooseframework.inl.gov/syntax/Positions/index.html

Guillaume

**Reply by lyyc199586** [reply]
I actually want to find a way to track the position of the crack tip in phase field and calculate the velocity in MOOSE. I know I can do it with python and/or paraview, but Positions object might be working for me

**Reply by hugary1995** [reply, marked as answer]
You may want to take look at this paper: https://onlinelibrary.wiley.com/doi/epdf/10.1002/nme.6879
If you want to pursue this I can get their source code for you.

Back to your original question, I think postprocessing the results externally probably is the right way to go, as it's more complicated than just tracking the coordinates of the nodes with phase-field == 1 even if the crack does not branch, unless you know the crack path a priori.

**Reply by lyyc199586** [reply]
I and Andre noticed this paper, and he wrote some python code on it.

I understand that if we want to track random unknown crack tips (like branching, etc), it  will be better to do it externally.  Here I am just trying to track crack tips for very simple cases, e.g. single crack. here is my workaround:

1. using a `ParsedAux` to calculate a distance field to the initial crack tip, say (0, 0): `if(d>d_cr, sqrt(x^2+y^2, -100)`, so that only x(d>d_cr) has a positive distance.
2. use `NodalMaxValueId` to get id of the node has largest distance
3. write a PP (like called NodalPosition..) similar to NodalVaribleValue, but just output the coordinates of this node.

I think this at least will work for single crack, enough for me to do some simple tests.

But yea, I agree we should move to an external solution finally, if you have their code about that paper that will be so nice!
